AmplifyTools/AmplifyCLIiOS/Sources/AmplifyCLIiOS/Core/Support/XcodeProject.swift

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import PathKit
+import XcodeProj
+
+enum XcodeProjectError: Error {
+    case notFound(path: String)
+    case noPbxProjFound
+    case groupNotFound(group: String)
+    case addFileFailed
+}
+
+enum XcodeProjectFileType {
+    case resource, source
+}
+
+struct XcodeProjectFile {
+    let path: String
+    let type: XcodeProjectFileType
+
+    init(_ path: String, type: XcodeProjectFileType) {
+        self.path = path
+        self.type = type
+    }
+}
+
+/// Wrapper around `XcodeProj` library, provides convenience utilities functions to
+/// access and update an Xcode project.
+struct XcodeProject {
+    let project: XcodeProj
+    let basePath: String
+    let projFilePath: String
+    let pbxProjFilePath: String
+
+    init(at path: String, projPath: String) throws {
+        self.basePath = path
+        self.project = try XcodeProj(pathString: projPath)
+        self.projFilePath = projPath
+        self.pbxProjFilePath = URL(fileURLWithPath: projPath).appendingPathComponent("project.pbxproj").path
+    }
+
+    func update() throws {
```

Comment:
Ah, now I see what `update` does. I think maybe `write()` or `save()` (or even `synchronize()` to be parallel with the old UserDefaults method) reflects the intent better

Reply:
I'll update to `synchronize()`